cmd/local-gadget/trace/dns.go

```diff
@@ -0,0 +1,164 @@
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+const logalGadgetSubKey = "local-gadget-key"
+
+func newDNSCmd() *cobra.Command {
+	var commonFlags utils.CommonFlags
+
+	// The DNS gadget works in a different way than most gadgets: It
+	// attaches a new eBPF program to each container when it's
+	// created instead of using an eBPF map with the mount
+	// namespaces IDs to filter the events. For this reason we can't
+	// use the TraceGadget implementation here.
+	runCmd := func(*cobra.Command, []string) error {
+		localGadgetManager, err := localgadgetmanager.NewManager(commonFlags.RuntimeConfigs)
+		if err != nil {
+			return commonutils.WrapInErrManagerInit(err)
+		}
+		defer localGadgetManager.Close()
+
+		// local-gadget is designed to trace containers, hence enable this column
+		cols := dnsTypes.GetColumns()
+		col, _ := cols.GetColumn("container")
+		col.Visible = true
+
+		parser, err := commonutils.NewGadgetParserWithRuntimeInfo(&commonFlags.OutputConfig, cols)
+		if err != nil {
+			return commonutils.WrapInErrParserCreate(err)
+		}
+
+		eventCallback := func(event dnsTypes.Event) {
+			baseEvent := event.GetBaseEvent()
+			if baseEvent.Type != eventtypes.NORMAL {
+				commonutils.ManageSpecialEvent(baseEvent, commonFlags.Verbose)
+				return
+			}
+
+			// TODO: This enrich logic could be moved to the
+			// gadget itself to simplify this here and in
+			// pkg/gadget-collection/gadgets/trace/dns/gadget.go
+			container := localGadgetManager.GetContainer(event.Key)
+			if container != nil {
+				event.Namespace = container.Namespace
+				event.Pod = container.Podname
+				event.Container = container.Name
+			}
+
+			switch commonFlags.OutputMode {
+			case commonutils.OutputModeJSON:
+				b, err := json.Marshal(event)
+				if err != nil {
+					fmt.Fprint(os.Stderr, fmt.Sprint(commonutils.WrapInErrMarshalOutput(err)))
+					return
+				}
+
+				fmt.Println(string(b))
+			case commonutils.OutputModeColumns:
+				fallthrough
+			case commonutils.OutputModeCustomColumns:
+				fmt.Println(parser.TransformIntoColumns(&event))
+			default:
+				fmt.Fprint(os.Stderr, commonutils.WrapInErrOutputModeNotSupported(commonFlags.OutputMode))
+			}
+		}
+
+		tracer, err := dnsTracer.NewTracer()
+		if err != nil {
+			return commonutils.WrapInErrGadgetTracerCreateAndRun(err)
+		}
+		defer tracer.Close()
+
+		if commonFlags.OutputMode != commonutils.OutputModeJSON {
+			fmt.Println(parser.BuildColumnsHeader())
+		}
+
+		netnsHost, _ := containerutils.GetNetNs(os.Getpid())
+		genKey := func(container *containercollection.Container) string {
+			if container.Netns == netnsHost {
+				return "host"
+			}
+			return container.ID
+		}
+
+		attachContainerFunc := func(container *containercollection.Container) {
+			err = tracer.Attach(genKey(container), container.Pid, eventCallback)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to trace container: %q: %s", container.Name, err)
```

Comment:
We have been adding `Error` prefix when printing errors directly:

```suggestion
				fmt.Fprintf(os.Stderr, "Error: starting tracing container %q: %s", container.Name, err)
```

Maybe we should have a helper function for this to help keep consistency.